ios/REANodesManager.mm

```diff
@@ -211,14 +204,16 @@ - (instancetype)initWithModule:(REAModule *)reanimatedModule uiManager:(RCTUIMan
     _viewRegistry = [_uiManager valueForKey:@"_viewRegistry"];
     _shouldFlushUpdateBuffer = false;
   }
+#endif
 
-  _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(onAnimationFrame:)];
-  _displayLink.preferredFramesPerSecond = 120; // will fallback to 60 fps for devices without Pro Motion display
-  [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
-  [_displayLink setPaused:true];
+  RCTExecuteOnMainQueue(^() {
+    _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(onAnimationFrame:)];
+    _displayLink.preferredFramesPerSecond = 120; // will fallback to 60 fps for devices without Pro Motion display
+    [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
+    [_displayLink setPaused:true];
+  });
```

Comment:
Similarly here, with an async init, couldn't it be possible to init this `REANodesManager` and immediately try to use `_displayLink` before it's initialized in something like `- (void)invalidate` or `- (void)postOnAnimation:(REAOnAnimationCallback)clb`?